### ISSUE TITLE
Trouble with ICommInfoReplyMsg 

### ISSUE THREAD

**Issue by TK-21st** [jupyterlab/jupyterlab#4639]
Tried checking status of a given comm as follows
```
this.session.kernel.requestCommInfo({'target_name': targetName})
      .then(msg=>{
        console.log(msg.content.status); // typescript gave error
      });
```
gave compilation error:
```bash
Property 'data' does not exist on type '{ comms: { [commId: string]: { 'target_name': string; }; }; }'.
```

Then tried
```typescript
this.session.kernel.requestCommInfo({'target_name': targetName})
      .then(msg=>{
        console.log(msg); // typescript gave error
      });
```

The returned message in the console has these keys `["header", "msg_id", "msg_type", "parent_header", "metadata", "content", "buffers", "channel"]`, and the `content` field has keys `["comms", "status"]` which has what I wanted. 

Tracked down the typescript issue and its here:
```typescript
/**
 * A `'comm_info_reply'` message on the `'stream'` channel.
 *
 * See [Messaging in Jupyter](https://jupyter-client.readthedocs.io/en/latest/messaging.html#comm-info).
 *
 * **See also:** [[ICommInfoRequest]], [[IKernel.commInfo]]
 */
interface ICommInfoReplyMsg extends IShellMessage {
    content: {
        /**
         * Mapping of comm ids to target names.
         */
        comms: {
            [commId: string]: {
                'target_name': string;
            };
        };
    };
}
```

**Comment by saulshanabrook** [comment]
So you wanna be able to access the `status` on the `content`?

**Comment by TK-21st** [comment]
Yep! I want to have it so that I can check if the `registerCommTarget` succeeded since it does not return a `Promise`. I've been having some errors in browser console that I think may be related to _when_ the `Comm` targets are registered.

### Problem Details:
 I followed the `Comm` [example](https://github.com/jupyterlab/jupyterlab/tree/master/packages/services), and wrote my code as follows:

```typescript
   this._kernelComm = this.session.kernel.registerCommTarget(this._commId, (comm, commMsg) => {
      if (commMsg.content.target_name !== this._commId) {
        return;
      }
      comm.onMsg = (msg) => {
        console.log(this._commId, msg.content.data);
      };
      comm.onClose = (msg) => {
        console.log(this._commId, msg.content.data); 
      };
    });
    
   //this._isCommOkay(this._commId); thinking about adding something like this here, which detects the status of the Comm target
  
    let code = [
      'from ipykernel.comm import Comm',
      '_comm = Comm(target_name="'+this._commId+ '")',
      '_comm.send(data="comm established")',
    ].join('\n');
    return this.session.kernel.requestExecute({ code: code }).done.then(() => {
      console.log('['+this.id+'] session started');
    });
```

All the messages get through without a problem even _without_ checking comm status, but for some reason I am getting this error in the browser console (`Dump-data` is my comm target name `this._commId`)
```
Uncaught (in promise) Error: Object 'Dump-data' not found in registry
```

**Comment by TK-21st** [comment]
Tracked down the source of the problem and it is related to how I am creating sessions and notebooks.

The way I am creating sessions and notebooks are as follows:
1. Spawn `ClientSession` with a given `path` and `kernel`
2. Register `Comm` target and create `Comm` in server side.
3. Using JLab's runtime application's `commands` to open/create the notebook file corresponding to the `path` in the previous step

The problem seems to be in step 3. What I _tried_ to do is to pass the `Kernel.IModel` from the spawned `session` directly to the `docmanager:open` command in the hope that the `docmanager` will use the _exact_ kernel model to open the document. However, because the document does not have a `context` registered (I created a `session` not a document), it goes through a context initialization process that seems to be spawning a new kernel (not sure about this). 

What is strange is that I can still access the server side `Comm` target from this new notebook (it is connected to the same kernel indeed), but every time I try sending something from the server side, a console error pops up that says the `Comm` object is not found in registry.

Closing this because the problem is from a different source.
